File: fixtures/throw.swift
func divide(a: Double, b: Double) throws -> Double {
  if b == 0.0 {
    throw LucentError(code: "DIVIDE_BY_ZERO", message: "Cannot divide by zero")
  }
  return (a / b)
}

func fail() throws -> Void {
  throw LucentError(code: "ALWAYS")
}

File: fixtures/throw.kt
fun divide(a: Double, b: Double): Double {
  if (b == 0.0) {
    throw LucentError("DIVIDE_BY_ZERO", message = "Cannot divide by zero")
  }
  return (a / b)
}

fun fail(): Unit {
  throw LucentError("ALWAYS")
}

fun failWithMetadata(path: String): Unit {
  throw LucentError("MISSING", message = "File\n不存在 🌍", metadata = mapOf("path" to path, "attempt" to 1.0, "retry" to false, "detail" to null))
}
